Start adding sitemap documentation (#766)
sharetribe/flex-docs: src/docs/ftw-analytics/template-sitemap/index.md

---
title: Sitemap in Sharetribe Web Template
slug: sitemap-in-template
updated: 2023-10-12
category: ftw-analytics
ingress:
  This guide describes how the default sitemap works in the Sharetribe
  Web Template, and how to customise it.
published: true
---

Your marketplace's sitemap helps search engines to process the correct
information on your marketplace. The Sharetribe Web Template has
built-in support for creating a sitemap (introduced in
[this change set](https://github.com/sharetribe/web-template/pull/243)),
and this article gives you some context into how sitemap generating
works in the template.

## Template sitemap structure

By default, the template generates several files related to the sitemap.
When running the template locally with _yarn run dev_, you can find
these routes on the dev server port, i.e. localhost:3500.

You can read more about the template sitemap and other resources in the
[Resources README](https://github.com/sharetribe/web-template/blob/main/server/resources/README.md).

### robots.txt

> The [your-domain]**/robots.txt** file details the authenticated routes
> that crawlers should avoid, and exposes the URL to the main sitemap. If

Reviewer: ```suggestion
The [your-domain]**/robots.txt** file details the authenticated routes
as disallowed for crawlers, and exposes the URL to the main sitemap. If
```

you custom develop routes that require authentication, it is a good idea
to specify them here.

### sitemap-index.xml

The [your-domain]**/sitemap-index.xml** file further exposes three
separate sub-sitemaps. This is useful because it allows you more
detailed tracking in analytics tools, and it reduces the size of each
individual sitemap.

> 

Reviewer: ```suggestion

This is the sitemap, you should give to [Google Search Console](https://www.sharetribe.com/help/en/articles/8413179-how-to-configure-google-search-console).

```

The link now leads to an article about how to configure the Google Search Console, but there might be a specific article written later on about how to feed the sitemap to Google Search Console.

Author: There is a section with an explanation and a link related to Google search console at the end of this article

listing URLs. If you make changes to the listing canonical URL
structure, be sure to make the corresponding changes to this sitemap as
well.

### sitemap-recent-pages.xml

The [your-domain]**/sitemap-recent-pages.xml** file shows the URLs for
your marketplace's asset-based pages that are shown in
_[your-domain]/p/:pathId_. The page information is fetched using the
_sdk.sitemapData.queryAssets()_ endpoint.

The default logic filters out the pages that have custom paths: landing
page, terms of service, and privacy policy. If you add public permanent
paths for asset-based pages to your template and your
_sitemap-default.xml_ file, be sure to filter them out of the SDK result
set in
[server/resources/sitemap.js](https://github.com/sharetribe/web-template/blob/main/server/resources/sitemap.js#L284)

## Sitemap caching

The template caches sitemap data
[for one day by default](https://github.com/sharetribe/web-template/blob/main/server/resources/sitemap.js#L55).
Especially on bigger sites with multiple listings, listing sitemap data
can take up to a few seconds to fetch from the SDK. Caching the results
improves the sitemap performance.

> 

Reviewer: Also, API will cache listings & pages endpoint. Currently, it's 1 day too, which means that the combined caching effect might be a maximum of 2 days on top of the sharetribe/web-template.
